Global Footer: Switch multiple nav blocks to lists inside a nav wrapper.

ryelle opened the pull request:
Fixes #402 — This switches to the global footer block to use a `nav` group block around 5 lists, instead of 5 separate nav blocks. Additionally, the label "Footer" has been added to the `nav`, so it now appears as "Footer navigation" when browsing by landmark.

This also had to change the way menu items were localized, removing the filter on nav-link blocks in favor of directly calling the function on appropriate links. 

Props @joedolson for the report

**Screenshots**

| | Production | Branch |
|---|---|---|
| Home, wide screen | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/e5ca0ff8-bfab-474e-9e96-eb1517963e8e) | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/e2bb35bf-2a41-4717-adbd-b9b9c10136c7) |
| Home, small screen | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/54ae9263-824d-4a61-af49-a5b24a1fbd6c) | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/00658dea-4a5f-4d5d-aec1-8d60686ee179) |
| Rosetta, wide screen | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/058af94b-ff48-4b1c-87f3-08a4eecf87e0) | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/849e5559-c540-471b-b8b5-cf97a400edb7) |
| Rosetta, small screen | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/284dfb73-c31b-40fe-9948-56f127beddcd) | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/022d4b0a-5f10-45e8-b539-d87813b912dc) |
| Classic theme, wide screen | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/814f6563-34c3-4668-abfe-1684232d8c64) | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/31130e79-5adc-4822-90cd-1ddc7fdba905) |
| Classic theme, small screen | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/76ae1998-b76d-4697-b293-91842d4e8823) | ![](https://github.com/WordPress/wporg-mu-plugins/assets/541093/bebee94e-db99-457a-b09e-b9a9ec9d1e21) |

**To test**

1. If you can, apply this to a sandbox and test on various sites— new & old themes, rosetta, etc.
2. Otherwise, use with one of the redesign sites (main, showcase, etc) to test locally
3. The footer is now one single `nav` element, individual items are lists
4. There should be no visual changes in the footer

Optionally test with a screen reader to see the single Footer navigation landmark.

ryelle committed:
Global Footer: Switch multiple nav blocks to lists inside a nav wrapper.

Fixes #402

ryelle committed:
Add an aria-label to the navigation

StevenDufresne reviewed (approved):
Looks good. Noting that there's a non-consequential `padding-right`.

```
.wp-block-group.global-footer>*, .wp-block-group.global-header {
    padding-right: var(--wp--custom--alignment--scroll-bar-width);
}
```

<img width="379" alt="Screenshot 2023-09-27 at 2 31 10 PM" src="https://github.com/WordPress/wporg-mu-plugins/assets/1657336/350772ae-e0cb-4204-81fe-f3717eac9d7e">

ryelle committed:
Global Footer: Remove scroll padding fix

This is not necessary on the footer items, because they are not full-width (the edge-spacing on the footer handles this).

ryelle commented:
Good catch, that doesn't seem to be necessary since the footer items are wrapped in a group with side-padding. I've removed it in https://github.com/WordPress/wporg-mu-plugins/pull/455/commits/3d3b06a36266f20ad38f48f1c543013b694e0db5.
